lms/djangoapps/mobile_api/course_dates/docs/0001-optimizing-performance-of-all-course-dates-api.rst

```diff
@@ -0,0 +1,64 @@
+Optimizing Performance of All Course Dates API
+=====================================================================
+
+Status
+------
+Proposed
+
+Context
+-------
+
+We encountered a performance issue while implementing an API for retrieving all course
+dates for a specific user. This API is required for the mobile application to display
+dates on the "Dates" tab, including the ability to shift due dates.
+
+After implementation we noticed significant performance degradation when users
+were enrolled in large courses containing numerous blocks. The function:
+
+.. code-block:: python
+
+    get_course_date_blocks(course, user, request, include_access=True, include_past_dates=True)
+
+took more than one second to return data for a single course in these cases.
```

Comment:
I think that either more details are needed here, like "for a course of X blocks, the execution time was Y" or vice versa, describe that the function inside goes through all the blocks of the course and passing them through transformers, so its use is very inefficient for writing courses.